uReplicator-Worker/src/main/scala/kafka/mirrormaker/CompactConsumerFetcherThread.scala

```diff
@@ -226,24 +247,44 @@ class CompactConsumerFetcherThread(name: String,
             isOOM = true
             if (consumerFetcherManager.systemExisting.compareAndSet(false, true)) {
               error("First OOM or processing error, call System.exit(-1);")
-              System.exit(-1);
+              System.exit(-1)
             }
             throw e
           }
         }
     }
   }
 
-  private def processFetchRequest(fetchRequest: FetchRequest) {
+  private def processFetchRequest(fetchRequestBuilder: FetchRequest.Builder) {
     val partitionsWithError = new mutable.HashSet[TopicAndPartition]
     var response: FetchResponse = null
     try {
-      trace("Issuing to broker %d of fetch request %s".format(sourceBroker.id, fetchRequest))
-      response = simpleConsumer.fetch(fetchRequest)
+      trace("Issuing to broker %d of fetch request builder %s".format(sourceBroker.id, fetchRequestBuilder))
+      response = simpleConsumer.fetch(fetchRequestBuilder)
     } catch {
+      case _: ConnectionFailedException =>
+        try {
+          warn("Issue with new simple consumer connection. Retrying...")
+          trace("Issuing to broker %d of fetch request builder %s".format(sourceBroker.id, fetchRequestBuilder))
+          simpleConsumer.reconnect()
```

Comment:
We need some backoff for reconnect.

Reply:
This is not an infinite loop, so the server shouldn't be heavily overwhelmed by repetitive connections